### DISCUSSION TITLE
is there a way Migrating User Name and password from Orchard cms 1.10 to OrchardCore

### DISCUSSION THREAD

**Discussion by yassinsayed** [OrchardCMS/OrchardCore#13290] [Q&A]
is there a way or what is the best way to migrate the username, email and password from orchard cms 1.10 to orchard core

**Comment by yassinsayed** [comment]
any help in this

**Reply by hishamco** [reply]
If I'm not wrong Orchard CMS already have import / export feature

**Reply by yassinsayed** [reply]
Yes it has, and we are able to export the users content, the issue is how ro import it to Orchard Core with the same password even without it

I hope to get support, we almost moved all our site contents and functions to orchard core the only remaining the users so we can go live

**Reply by hishamco** [reply]
Frankly I didn't work too much in Orchard CMS, but simply you can export a sample users from both platform, then you can tweak the exported file from Orhard CMS to match the reipce schema for OC

Let me ping Lombiq guys if they did something before  /cc @Piedone @BenedekFarkas @domonkosgabor

**Reply by yassinsayed** [reply]
no export step in orchard core to export users, Please advise how to export users from orchard core to tweak it with the exported orchard cms

**Reply by hishamco** [reply]
You can easily add it if you have the source code

**Comment by Piedone** [comment]
There's no easy way to do this. My approach would be the following:

1. Recreate users in OC from a recipe. For this, you can generate [`createUser` commands](https://github.com/OrchardCMS/OrchardCore/blob/a04a6bebc606e81c4d8cd21f8186b2b20ccfb0c0/src/OrchardCore.Modules/OrchardCore.Users/Commands/UserCommands.cs#L34) from the O1 site. Since you won't be able to set their passwords to their original ones, use a 64 character random password for everyone.
2. Extend the OC login so if login fails for a user account coming from O1, they get redirected to a page telling them why they can't log in and point them to reset their password. You can do this by hooking into `ILoginFormEvent.LoggingInFailedAsync()`. To make it sure that you only show this page to users who haven't done this first password reset yet (as opposed to those who've done it but can't log in due to a newly mistyped password) you can extend the User entity with some metadata that stores whether the user is an originally OC one, and whether they've already done this first password reset. You can look into how we handle Book entities [here](https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/dev/Controllers/DatabaseStorageController.cs); you need to do something similar with `User`.
3. Your users will be gradually migrated.

**Comment by yassinsayed** [comment]
thanks @Piedone will work on it
